pkg/cmd/cluster-policy-controller/cmd.go

```diff
@@ -1,115 +1,300 @@
 package cluster_policy_controller
 
 import (
+	"context"
 	"fmt"
-	"io"
 	"io/ioutil"
+	"math/rand"
 	"os"
-	"path"
+	"path/filepath"
+	"time"
 
-	configv1 "github.com/openshift/api/config/v1"
-	openshiftcontrolplanev1 "github.com/openshift/api/openshiftcontrolplane/v1"
-	"github.com/openshift/library-go/pkg/config/helpers"
-	"github.com/openshift/library-go/pkg/serviceability"
 	"github.com/spf13/cobra"
-	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/serializer"
-	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	"k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/apiserver/pkg/server"
+	"k8s.io/component-base/logs"
 	"k8s.io/klog/v2"
+
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	"github.com/openshift/library-go/pkg/config/client"
+	"github.com/openshift/library-go/pkg/config/configdefaults"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/openshift/library-go/pkg/controller/fileobserver"
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/serviceability"
+
+	// for metrics
+	_ "github.com/openshift/library-go/pkg/controller/metrics"
+
+	clusterpolicyversion "github.com/openshift/cluster-policy-controller/pkg/version"
 )
 
 const RecommendedStartControllerName = "cluster-policy-controller"
 
-type ClusterPolicyController struct {
-	ConfigFilePath string
-	// KubeConfigFile points to a kubeconfig file if you don't want to use the in cluster config
-	KubeConfigFile string
+func NewClusterPolicyControllerCommand(name string) *cobra.Command {
+	cmd := NewClusterPolicyControllerCommandConfig("cluster-policy-controller", clusterpolicyversion.Get(), RunClusterPolicyController).
+		NewCommandWithContext(context.TODO())
```

Comment:
`context.Background()` ?

Reply:
I am little bit confused by ` It is never canceled`: https://golang.org/pkg/context/#Background . Nevertheless, I updated the code.